optimum/onnx/transformations_utils.py

```diff
@@ -45,13 +46,19 @@ def _find_duplicate_initializers(
                     if tensor_data:
                         tensor_data = tuple(tensor_data)
                         break
-                duplicates[(initializer.data_type, tensor_data, tensor_dims)].add((initializer.name, i))
+
+                # Hash tensor data to avoid storing large amounts of data in memory
+                hashed = hashlib.sha512()
+                hashed.update(str(tensor_data).encode("utf-8"))
```

Comment:
A small nit because I think it actually never happens but could we do a check that `tensor_data` is None?